### ISSUE TITLE
0.4.10.2 Compile errors on Windows

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#3534]
Bug Report
==========

Environment
-----------

* **FreeOrion Version**: 0.4.10.2 1d5919ffc8516343ff96538e0999b5fc17621d3d
* **Operating System**: Windows 10
* **Fetched as** 
  * Compiled from source

Description
-----------

When I tried to build RC1 for the 0.4.10.2 maintenance release, I encountered the following compile errors on Windows:
```
...

5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3031): error C2429: language feature 'structured bindings' requires compiler flag '/std:c++17'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3035): error C2065: 'gifted_objects': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3035): error C3312: no callable 'begin' function found for type 'unknown-type'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3035): error C3312: no callable 'end' function found for type 'unknown-type'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3036): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3040): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3042): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3043): error C3536: 'system': cannot be used before it is initialized
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3051): error C3536: 'sys_it': cannot be used before it is initialized
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3051): error C2679: binary '!=': no operator found which takes a right-hand operand of type 'std::_Tree_iterator<std::_Tree_val<std::_Tree_simple_types<_Ty>>>' (or there is no acceptable conversion)
5>        with
5>        [
5>            _Ty=std::pair<const int,bool>
5>        ]

...

5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3057): error C2065: 'system_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3057): error C2065: 'recipient_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3070): error C2065: 'recipient_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3070): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3075): error C2429: language feature 'structured bindings' requires compiler flag '/std:c++17'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3076): error C2065: 'initial_recipient_ids': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3076): error C2429: language feature 'structured bindings' requires compiler flag '/std:c++17'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3077): error C2065: 'gifted_objects': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3077): error C3312: no callable 'begin' function found for type 'unknown-type'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3077): error C3312: no callable 'end' function found for type 'unknown-type'
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3078): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3079): error C2065: 'contained_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3079): error C2065: 'initial_owner_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3080): error C2065: 'contained_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3080): error C2065: 'recipient_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3082): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3082): error C2065: 'recipient_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3084): error C2065: 'recipient_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3084): error C3536: 'empire': cannot be used before it is initialized
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3085): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3086): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3086): error C2065: 'initial_owner_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3087): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3088): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3088): error C2065: 'initial_owner_empire_id': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3091): error C2065: 'gifted_obj': undeclared identifier
5>c:\users\user\softwareprojekte\fo\release\freeorion\server\serverapp.cpp(3091): error C2065: 'recipient_empire_id': undeclared identifier

...
```
I'm using MSVC 2017 and version 11 of the FO SDK.

Judging by the error messages, apparently one of the backported fixes requires C++17, which isn't enabled on the 0.4.10 release branch. Enabling it leads to other compile errors which would have to be fixed, and this kind of mess is exactly what we wanted to avoid for the 0.4.10.2 maintenance release.

One solution could be to remove the problematic backport.

How to proceed? - @geoffthemedio?

**Comment by geoffthemedio** [comment]
replacing structured bindings is fairly simple... try with https://github.com/freeorion/freeorion/tree/nocpp17

**Comment by Vezzra** [comment]
Yep, that worked.

**Comment by Vezzra** [comment]
Fixed by f663dad6379658b539bd28c95229218eb4333ef2
